torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.cpp

```diff
@@ -17,171 +18,881 @@ namespace fuser {
+        // and outer partial extent:
+        //   auto outer_partial_extent =
+        //   SimplifyingIrBuilder::divExpr(getMaybePartialMappedExtent(merge->out()),
+        //   merge->inner()->extent())
+        //
+        // However, for right now we'll assume no partial outer dimension maps
+        // unless the extent of merge->out and merge->inner are compile time
+        // constant. If not compile time constant we'll take:
+        //   min(merge->inner()->extent(), the partial extent of merge->out())
+        // for the inner partial map, and assume outer doesn't partially map.
+        //
+        // TODO: Improve to do the above. The challenge here is adding correct
+        // conditionals to evaluate (and add conditional evaluation in expr
+        // evaluator) to check vectorization especially like (out_extent >
+        // inner_extent).
+        if (!getMaybePartialMappedExtent(merge->out())->isConstInt() ||
+            !merge->inner()->extent()->isConstInt()) {
+          // We don't know at compile time if the partial extent of merge->out
+          // is bigger than merge->inner or not, so generate an expressions we
+          // can evaluate at runtime for the inner partial mapping extent. Don't
+          // attempt to map merge->outer, just assume it doesn't partially map.
+          ids[out_pos] = merge->inner();
+          partial_projected_extent_[merge->inner()] =
+              SimplifyingIrBuilder::minExpr(
+                  getMaybePartialMappedExtent(merge->out()),
+                  merge->inner()->extent());
+        } else {
+          // Compile time constant analysis
+          auto partial_out_extent =
+              getMaybePartialMappedExtent(merge->out())->evaluateInt();
+          auto inner_extent = merge->inner()->extent()->evaluateInt();
+          if (partial_out_extent > inner_extent) {
+            // Partial mapping on out is bigger than inner, so extend partial
+            // mapping to outer and mark inner as fully mapped.
+            //
+            // TODO: Should this be ceilDiv? Normal div is more conservative, so
+            // sticking with that for now.
+            partial_projected_extent_[merge->outer()] =
+                SimplifyingIrBuilder::divExpr(
+                    getMaybePartialMappedExtent(merge->out()),
+                    merge->inner()->extent());
+            // Map inner and outer dimensions
+            ids[out_pos] = merge->outer();
+            ids.insert(ids.begin() + out_pos + 1, merge->inner());
+          } else if (partial_out_extent == inner_extent) {
+            // Only map inner dimension, but since it's the same extent as the
+            // partial out extent, map all of it.
+            ids[out_pos] = merge->inner();
+          } else {
+            // Partially map inner dimension
+            ids[out_pos] = merge->inner();
+            partial_projected_extent_[merge->inner()] =
+                getMaybePartialMappedExtent(merge->out());
```

Comment:
Similar to https://github.com/csarofeen/pytorch/pull/2124#discussion_r1004991114 and https://github.com/csarofeen/pytorch/pull/2124/files#r1004993740, I think we should check divisibility here.